internal/config/cgmgr/cgmgr.go

```diff
@@ -0,0 +1,80 @@
+// +build linux
+
+package cgmgr
+
+import (
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/pkg/errors"
+)
+
+const (
+	scopePrefix = "crio"
+	// minMemoryLimit is the minimum memory that must be set for a container.
+	// A lower value would result in the container failing to start.
+	// this value has been arrived at for runc on x86_64 hardware
+	minMemoryLimit = 12 * 1024 * 1024
+	// CgroupfsCgroupManager represents cgroupfs native cgroup manager
+	cgroupfsCgroupManager = "cgroupfs"
+	// SystemdCgroupManager represents systemd native cgroup manager
+	systemdCgroupManager = "systemd"
+
+	DefaultCgroupManager = systemdCgroupManager
+
+	// these constants define the path and name of the memory max file
+	// for v1 and v2 respectively
+	cgroupMemoryPathV1    = "/sys/fs/cgroup/memory"
+	cgroupMemoryMaxFileV1 = "memory.limit_in_bytes"
+	cgroupMemoryPathV2    = "/sys/fs/cgroup"
+	cgroupMemoryMaxFileV2 = "memory.max"
+)
+
+// CgroupManager is an interface to interact with cgroups on a node. CRI-O is configured at startup to either use
+// systemd or cgroupfs, and the node itself is booted with cgroup v1, or cgroup v2. CgroupManager is an interface for
+// the CRI-O server to use cgroups, regardless of how it or the node was configured.
+type CgroupManager interface {
+	// String returns the name of the cgroup manager (either cgroupfs or systemd)
+	Name() string
+	// IsSystemd returns whether it is a systemd cgroup manager
+	IsSystemd() bool
+	// GetContainerCgroupPath takes arguments sandbox parent cgroup and container ID and returns
+	// the cgroup path for that containerID. If parentCgroup is empty, it
+	// uses the default parent for that particular manager
+	GetContainerCgroupPath(string, string) string
```

Comment:
"Effective Go" recommends to drop the `Get` prefix. https://golang.org/doc/effective_go.html#Getters